src/components/ColorSelect.jsx

```diff
@@ -0,0 +1,24 @@
+import { useQRCodeGenerator } from "../hooks/useQRCodeGenerator"
+
+export const  ColorSelect = () => {
+
+    const { color, setColor } = useQRCodeGenerator();
```

Comment:
I read that you where experiencing issues with having the color selector in a separate component? I had the same issue with parts of my code. I'm not entirely sure, but perhaps you could try adding the handleColorChange in the hook, and then importing that also?